webknossos-datastore/app/com/scalableminds/webknossos/datastore/services/ManagedS3Service.scala

```diff
@@ -25,24 +29,44 @@ import scala.concurrent.ExecutionContext
 import scala.jdk.CollectionConverters._
 import scala.jdk.FutureConverters._
 
-class ManagedS3Service @Inject()(dataStoreConfig: DataStoreConfig) extends FoxImplicits with LazyLogging {
+class ManagedS3Service @Inject()(config: DataStoreConfig) extends FoxImplicits with LazyLogging {
 
-  private lazy val s3UploadCredentialsOpt: Option[(String, String)] =
-    dataStoreConfig.Datastore.DataVaults.credentials.flatMap { credentialConfig =>
+  def pathIsInManagedS3(path: UPath): Boolean =
+    path.getScheme.contains(PathSchemes.schemeS3) && globalCredentials.exists(c =>
+      UPath.fromString(c.name).map(path.startsWith).getOrElse(false))
+
+  def findGlobalCredentialFor(pathOpt: Option[UPath]): Option[S3AccessKeyCredential] =
+    pathOpt.flatMap(findGlobalCredentialFor)
```

Comment:
just a suggestion: Maybe move this next to the other findGlobalCredentialFor implementation?